store/router.go

```diff
@@ -117,6 +119,10 @@ func (r *Router) Get(ctx context.Context, key []byte, cm commitments.CommitmentM
 
 // Put ... inserts a value into a storage backend based on the commitment mode
 func (r *Router) Put(ctx context.Context, cm commitments.CommitmentMode, key, value []byte) ([]byte, error) {
+	if r.readonly {
+		return nil, errors.New("read-only router does not support Put")
+	}
```

Comment:
what if I just wanna run proxy for S3 support using the `keccak256` backend and would like to disable reading from eigenda?

Reply:
I'm a bit confused, the read-only mode implemented here will disable all `put` methods no matter what. Are you saying the read-only mode should handle something more granular with the cache/fallback targets?